python/cuml/test/test_random_forest.py

```diff
@@ -237,3 +236,105 @@ def test_rf_regression_default(datatype, column_info, nrows):
     except AssertionError:
         pytest.xfail("failed due to AssertionError error, "
                      "fix will be merged soon")
+
+
+@pytest.mark.parametrize('datatype', [np.float32])
+@pytest.mark.parametrize('column_info', [unit_param([16, 7]),
+                         quality_param([200, 100]),
+                         stress_param([500, 350])])
+@pytest.mark.parametrize('nrows', [unit_param(500), quality_param(5000),
+                         stress_param(500000)])
+def test_rf_regression_seed(datatype, column_info, nrows):
+
+    ncols, n_info = column_info
+    X, y = make_regression(n_samples=100, n_features=ncols,
+                           n_informative=n_info,
+                           random_state=123)
+    X = X.astype(datatype)
+    y = y.astype(datatype)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8,
+                                                        random_state=0)
+    for i in range(20):
+        seed = random.randint(100, 1e5)
+        # Initialize, fit and predict using cuML's
+        # random forest classification model
+        cu_reg = curfr(seed=seed, n_streams=1)
+        cu_reg.fit(X_train, y_train)
+
+        # predict using FIL
+        fil_preds_orig = cu_reg.predict(X_test,
+                                        predict_model="GPU").copy_to_host()
+        cu_preds_orig = cu_reg.predict(X_test,
+                                       predict_model="CPU")
+        cu_r2_orig = r2_score(y_test, cu_preds_orig, convert_dtype=datatype)
+        fil_r2_orig = r2_score(y_test, fil_preds_orig, convert_dtype=datatype)
+
+        # Initialize, fit and predict using cuML's
+        # random forest classification model
+        cu_reg2 = curfr(seed=seed, n_streams=1)
+        cu_reg2.fit(X_train, y_train)
+
+        # predict using FIL
+        fil_preds_rerun = cu_reg2.predict(X_test,
+                                          predict_model="GPU").copy_to_host()
+        cu_preds_rerun = cu_reg2.predict(X_test, predict_model="CPU")
+        cu_r2_rerun = r2_score(y_test, cu_preds_rerun, convert_dtype=datatype)
+        fil_r2_rerun = r2_score(y_test, fil_preds_rerun,
+                                convert_dtype=datatype)
+        try:
+            assert array_equal(fil_r2_orig, fil_r2_rerun, tol=1e-3)
+            assert array_equal(fil_preds_orig, fil_preds_rerun, tol=1e-3)
+            assert array_equal(cu_r2_orig, cu_r2_rerun, tol=1e-3)
+            assert array_equal(cu_preds_orig, cu_preds_rerun, tol=1e-3)
+
+        except AssertionError:
+            pytest.xfail("failed due to AssertionError error as setting the"
+                         " seed in not creating reproducible results")
```

Comment:
Same as above, this try/except totally defeats the purpose of the test. This one seems like it might be appropriate to make it xfail. And create/link the appropriate github issue.